src/main/java/uk/gov/companieshouse/web/accounts/transformer/smallfull/impl/CapitalAndReservesTransformerImpl.java

```diff
@@ -0,0 +1,203 @@
+
+        // Other reserves
+        if (capitalAndReservesApi.getOtherReserves() != null) {
+            OtherReserves otherReserves = createOtherReserves(balanceSheet);
+            otherReserves.setCurrentAmount(capitalAndReservesApi.getOtherReserves());
+        }
+
+        // Profit and Loss
+        if (capitalAndReservesApi.getProfitAndLoss() != null) {
+            ProfitAndLossAccount profitAndLossAccount = createProfitAndLossAccount(balanceSheet);
+            profitAndLossAccount.setCurrentAmount(capitalAndReservesApi.getProfitAndLoss());
+        }
+
+        // Share premium accounts
+        if (capitalAndReservesApi.getSharePremiumAccount() != null) {
+            SharePremiumAccount sharePremiumAccount = createSharePremiumAccounts(balanceSheet);
+            sharePremiumAccount.setCurrentAmount(capitalAndReservesApi.getSharePremiumAccount());
+        }
+
+        // Shareholder funds
+        if (capitalAndReservesApi.getTotalShareholdersFunds() != null) {
+            TotalShareholdersFunds totalShareholdersFunds = createTotalShareholdersFund(balanceSheet);
+            totalShareholdersFunds.setCurrentAmount(capitalAndReservesApi.getTotalShareholdersFunds());
+        }
+    }
+
+    @Override
+    public void addPreviousPeriodToWebModel(BalanceSheet balanceSheet, BalanceSheetApi balanceSheetApi) {
+
+        createCapitalAndReserves(balanceSheet);
+        CapitalAndReservesApi capitalAndReservesApi = balanceSheetApi.getCapitalAndReservesApi();
+
+        // Share capital
+        if (capitalAndReservesApi.getCalledUpShareCapital() != null) {
+            CalledUpShareCapital calledUpShareCapital = createCalledUpShareCapital(balanceSheet);
+            calledUpShareCapital.setPreviousAmount(capitalAndReservesApi.getCalledUpShareCapital());
+        }
+
+        // Other reserves
+        if (capitalAndReservesApi.getOtherReserves() != null) {
+            OtherReserves otherReserves = createOtherReserves(balanceSheet);
+            otherReserves.setPreviousAmount(capitalAndReservesApi.getOtherReserves());
+        }
+
+        // Profit and Loss
+        if (capitalAndReservesApi.getProfitAndLoss() != null) {
+            ProfitAndLossAccount profitAndLossAccount = createProfitAndLossAccount(balanceSheet);
+            profitAndLossAccount.setPreviousAmount(capitalAndReservesApi.getProfitAndLoss());
+        }
+
+        // Share premium accounts
+        if (capitalAndReservesApi.getSharePremiumAccount() != null) {
+            SharePremiumAccount sharePremiumAccount = createSharePremiumAccounts(balanceSheet);
+            sharePremiumAccount.setPreviousAmount(capitalAndReservesApi.getSharePremiumAccount());
+        }
+
+        // Shareholder funds
+        if (capitalAndReservesApi.getTotalShareholdersFunds() != null) {
+            TotalShareholdersFunds totalShareholdersFunds = createTotalShareholdersFund(balanceSheet);
+            totalShareholdersFunds.setPreviousAmount(totalShareholdersFunds.getPreviousAmount());
```

Comment:
The `setPreviousAmount()` call here is setting the value taken from the web model (i.e. `totalShareholdersFunds.getPreviousAmount()`) which will be 0 at this point. This should read:

```
totalShareholdersFunds.setPreviousAmount(capitalAndReservesApi.getTotalShareholdersFunds());
```

Reply:
Thanks @marcransome  - resolved in 64b88e8